src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,39 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden = false) {
+    super(name, health);
+
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(target) {
+    if (target.hasOwnProperty('hidden')) {
+      if (!target.hidden) {
+        target.health -= 50;
+      }
+    }
```

Comment:
```suggestion
  bite(target) {
    if (targer instanceof Herbivore && !target.hidden) {
      target.health -= 50;
    }
```

Same but a bit shorter